tests/PrepareBodyMiddlewareTest.php

```diff
@@ -73,6 +73,24 @@ function (RequestInterface $request) {
         $this->assertEquals(200, $response->getStatusCode());
     }
 
+    public function testAddsContentLengthForGET()
```

Comment:
I think we could reuse the existing test (maybe with a data provider) and instead add test for the case when there is no body (Content-Length should not be added then).